src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py

```diff
@@ -275,6 +283,7 @@ def setup_slots(self):
         self.model.itemChanged.connect(self.model_changed)
 
         self.trigger.connect(self.finish_transform)
+        self.txtBackground.textChanged.connect(self.on_background_text_changed)
```

Comment:
Wouldn't `editingChanged` signal be better here? We just need to assure the final value is correct, not respond to every keystroke?
However, I can see a lot of artistic freedom with both signals in the code, so this may just slide. Or we can decide on certain behaviour and mandate it.

Reply:
You mean `editingFinished`? Personally, I like the current behaviour, it changes colour as soon as you type something wrong.

Reply:
`editingFinished` of course.
Alternatively, one might just add a validator to the line edit, like we do in a number of other places. This will limit the characters which can be entered in the line to only valid ones.
I can still enter silly strings in the background field (or any other line edit in this perspective) and it will happily accept it, even if the bg turns yellow.
The helpful message in the log tells me what's wrong but it would probably be better to just use the validators, like in all other perspectives.

Reply:
All the QLineEdits in corfunc need validators or need to be read only, I'll create a separate issue for this. The yellow background was the feature request, and it's more to let the user know that they have calculated or entered a negative number than it is to stop them doing so.

Reply:
See issue #2129